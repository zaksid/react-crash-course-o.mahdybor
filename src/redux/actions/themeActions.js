import Constants from '../constants';

export const setResultsTheme = theme => ({
    type: Constants.SetResultsTheme,
    payload: theme
});

export const setToolbarTheme = theme => ({
    type: Constants.SetToolbarTheme,
    payload: theme
});
