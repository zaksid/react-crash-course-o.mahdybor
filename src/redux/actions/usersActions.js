import Constants from '../constants';

export const fetchUsersStart = () => ({
    type: Constants.FetchUserStart
});

export const usersLoaded = users => ({
    type: Constants.FetchUserSuccess,
    payload: users
});

export const fetchCancel = () => ({
    type: Constants.FetchUserCancel
});

export const fetchError = error => ({
    type: Constants.FetchUserError,
    payload: error
});
