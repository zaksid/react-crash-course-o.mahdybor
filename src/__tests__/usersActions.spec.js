import { setResultsTheme } from '../redux/actions/themeActions';
import { fetchCancel, fetchError, fetchUsersStart, usersLoaded } from '../redux/actions/usersActions';

describe('setResultsTheme', () => {
    const actionFetchUsersStart = fetchUsersStart();
    const actionUsersLoaded = usersLoaded([]);
    const actionFetchCancel = fetchCancel();
    const actionFetchError = fetchError('error');

    it('all should return correct action type', () => {
        expect(actionFetchCancel.type).toBe('FETCH_USERS_CANCEL');
        expect(actionFetchError.type).toBe('FETCH_USERS_ERROR');
        expect(actionFetchUsersStart.type).toBe('FETCH_USERS_START');
        expect(actionUsersLoaded.type).toBe('FETCH_USERS_SUCCESS');

    });

    it('all should match snapshot', () => {
        expect(actionFetchCancel).toMatchSnapshot();
        expect(actionFetchError).toMatchSnapshot();
        expect(actionFetchUsersStart).toMatchSnapshot();
        expect(actionUsersLoaded).toMatchSnapshot();
    });
});
