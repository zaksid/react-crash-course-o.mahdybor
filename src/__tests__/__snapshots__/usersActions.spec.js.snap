// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test user actions all should match snapshot 1`] = `
Object {
  "type": "FETCH_USERS_CANCEL",
}
`;

exports[`Test user actions all should match snapshot 2`] = `
Object {
  "payload": "error",
  "type": "FETCH_USERS_ERROR",
}
`;

exports[`Test user actions all should match snapshot 3`] = `
Object {
  "type": "FETCH_USERS_START",
}
`;

exports[`Test user actions all should match snapshot 4`] = `
Object {
  "payload": Array [],
  "type": "FETCH_USERS_SUCCESS",
}
`;
